opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1366,6 +1366,8 @@ namespace Opm {
         // if a well or group change control it affects all wells that are under the same group
         for (const auto& well : well_container_) {
             well->updateWellStateWithTarget(ebosSimulator_, this->groupState(), this->wellState(), deferred_logger);
+            well->updatePrimaryVariables(this->wellState(), deferred_logger);
```

Comment:
It was found that those added `updatePrimaryVariables` and `initPrimaryVariablesEvaluation` are not totally necessary because they might be updated somewhere else. At the same time, the usage of these functions are very difficult to find where they are actually updated. Some refactoring should be needed to regulate the usage of these two functions.

Reply:
Testing shows that we can not remove all of them. Some added ones have some effects and work, while not easy to tell which ones. Will investigate.